Glagow| ITP-MAY-2025 | Radwan Alshujaa | Form control

Radwan-Alshujaa opened the pull request:
<!--

You must title your PR like this:

REGION | COHORT_NAME | FIRST_NAME LAST_NAME | PROJ_NAME

For example,

London | May-2025 | Carol Owen | Wireframe

Complete the task list below this message.
If your PR is rejected, check the task list.

-->

## Learners, PR Template

Self checklist

- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with REGION | COHORT_NAME | FIRST_NAME LAST_NAME | PROJ_NAME
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)
- [x] My changes meet the [requirements](./README.md) of this task

## Changelist

This PR implements the requested T-shirt order form as per the project requirements. The form first verifies the user by asking for their name and email, then collects their preferred T-shirt color and size. This update ensures the form meets both the functional and design specifications.Briefly explain your PR.

## Questions

Ask any questions you have for your reviewer.

cjyuan reviewed (commented):
I saw quite some good suggestions when I asked ChatGPT this:
```
review HTML code:
...        <--- paste your HTML code here
```

To get better at using AI tools, try running your code by ChatGPT. Take a look at the suggestions it gives you, and pick out the ones that actually make sense for your project. It's a great way to learn and improve your work.

cjyuan commented:
On a separate note, it is best practice to include a brief description of the changes you made in the Changelist section of every PR.

![image](https://github.com/user-attachments/assets/e6a68545-81bd-45ad-9586-db6ece688839)

cjyuan commented:
A well-prepared PR makes it easier for reviewers to approve it with minimal back-and-forth.
The "PR Essentials" file on the [Canvas](https://codeyourfutur-yov6609.slack.com/archives/C07QX99JK7B) in the #cyf-code-review channel has some helpful tips to make your PRs more robust and ready for review.
Can you take a few minutes to check against the essentials?

After taking a quick look at your code, I noticed some errors in your HTML code, and some inconsistency on your HTML page.

Once you've finished making changes, please leave a comment on this PR so I’ll get notified. Thanks!

Radwan-Alshujaa commented:
I improved the form by adding fieldset and legend for better accessibility, included value attributes for radio buttons, corrected the dropdown placeholder, and added helpful instructions and an ARIA label for the submit button.

cjyuan commented on Form-Controls/index.html:
```diff
@@ -1,27 +1,76 @@
+      <!-- required radial color input -->
+      <div>
+       <fieldset>
+  <legend>Choose a color:</legend>
+  <div>
+    <input type="radio" id="red" name="color" value="red" required />
+    <label for="red">Red</label>
+  </div>
+  <div>
+    <input type="radio" id="green" name="color" value="green" />
+    <label for="green">Green</label>
+  </div>
+  <div>
+    <input type="radio" id="black" name="color" value="black" />
+    <label for="black">Black</label>
+  </div>
+</fieldset>
```
The indentation of these few lines of code seems off. 

You can try using VSCode "Format Document" feature to quickly format your code for better readability and consistency. To use the feature, right-click inside the code editor and select the option.
Please note that if there are syntax errors in the code, the "Prettier" extension may not format HTML code properly.

cjyuan reviewed (commented):
Code is error free. Form looks great! 

I think you can further improve your implementation by addressing these issues:
1. Lighthouse accessibility score is very high but not yet 100. It indicates, "Touch targets do not have sufficient size or spacing." Can you make necessary changes to improve the score to 100? ChatGPT can probably help. 

2. A user can currently enter a name consisting of only spaces (e.g., " "). To enforce stricter validation, consider using the `pattern` attribute to disallow whitespace-only name.

Radwan-Alshujaa commented:
I formated the code with  'Prettier'   and I added the pattern attribute to unsure the user will enter a valid name

cjyuan commented:
Changes look good! Well done!
